internal/pkg/server/agent.go

```diff
@@ -71,8 +71,8 @@ func NewAgent(cliCfg *ucfg.Config, reader io.Reader, bi build.Info, reloadables
 		reloadables: reloadables,
 	}
 	a.agent, _, err = client.NewV2FromReader(reader, client.VersionInfo{
-		Name:    kFleetServer,
-		Version: bi.Version,
+		Name:      kFleetServer,
+		BuildHash: bi.Commit,
```

Comment:
We replaced the version with the commit, because with the early agent releases project the version may not be unique anymore but the commit always is.